File: app/src/main/java/com/dertyp7214/appstore/components/BlackLine.java
/*
 *
 *  * Copyright (c) 2018.
 *  * Created by Josua Lengwenath
 *
 */

package com.dertyp7214.appstore.components;

import android.content.Context;
import android.graphics.Color;
import android.os.Build;
import android.util.AttributeSet;
import android.view.View;

import javax.annotation.Nullable;

public class BlackLine extends View {
    public BlackLine(Context context) {
        super(context);
        init();
    }

    public BlackLine(Context context, @Nullable AttributeSet attrs) {
        super(context, attrs);
        init();
    }

    public BlackLine(Context context, @Nullable AttributeSet attrs, int defStyleAttr) {
        super(context, attrs, defStyleAttr);
        init();
    }

    public BlackLine(Context context, @Nullable AttributeSet attrs, int defStyleAttr, int defStyleRes) {
        super(context, attrs, defStyleAttr, defStyleRes);
        init();
    }

    private void init() {
        if (Build.VERSION.SDK_INT >= 28) {
            this.setBackgroundColor(Color.parseColor("#e7e7e7"));
        }
    }
}

File: app/src/main/java/com/dertyp7214/appstore/components/BlackLine.kt
/*
 *
 *  * Copyright (c) 2018.
 *  * Created by Josua Lengwenath
 *
 */

package com.dertyp7214.appstore.components

import android.content.Context
import android.graphics.Color
import android.os.Build
import android.util.AttributeSet
import android.view.View

class BlackLine : View {
    constructor(context: Context) : super(context) {
        init()
    }

    constructor(context: Context, attrs: AttributeSet?) : super(context, attrs) {
        init()
    }

    constructor(context: Context, attrs: AttributeSet?, defStyleAttr: Int) : super(context, attrs, defStyleAttr) {
        init()
    }

    private fun init() {
        if (Build.VERSION.SDK_INT >= 28) {
            this.setBackgroundColor(Color.parseColor("#e7e7e7"))
        }
    }
}
